File: bankingtool.client/src/app/dto/request/bank-account/cash-withdraw-request.ts
import { RequestId } from "../request-id";

export interface CashWithdrawRequestObject {
  request: CashWithdrawRequest;
}

export interface CashWithdrawRequest {
  account: CashWithdrawRequestAccount;
  transaction: CashWithdrawRequestTransaction;
}

export interface CashWithdrawRequestAccount extends RequestId {

}

export interface CashWithdrawRequestTransaction {
  amount: number;
  description: string;
}

export const CreateCashWithdrawRequestObject = (accountId: number, amount: number, description: string): CashWithdrawRequestObject => {
  let model: CashWithdrawRequestObject = {
    request: {
      account: {
        id: accountId,
      },
      transaction: {
        amount: amount,
        description: description
      }
    }
  }
  return model;
}

File: BankingTool.Model/Dto/RequestDto/BankAccount/CashWithdrawRequest.cs
using BankingTool.Model.Dto.RequestDtos;

namespace BankingTool.Model.Dto.RequestDto.BankAccount
{
    public class CashWithdrawRequestObject
    {
        public CashWithdrawRequest Request { get; set; }
    }
    public class CashWithdrawRequest
    {
        public CashWithdrawRequestAccount Account { get; set; }
        public CashWithdrawRequestTransaction Transaction { get; set; }
    }
    public class CashWithdrawRequestAccount : RequestId
    {
    }
    public class CashWithdrawRequestTransaction
    {
        public int Amount { get; set; }
        public string Description { get; set; }
    }
}
